miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,73 @@
+use crate::assembler::assembler;
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::{dsa::rpo_falcon512, merkle::MerkleStore},
+    utils::vec,
+    AccountError, Word, ZERO,
+};
+
+pub enum AuthScheme {
+    RpoFalcon512 { pub_key: rpo_falcon512::PublicKey },
+}
+
+/// Creates a new faucet account with basic faucet interface, specified authentication scheme,
+/// and provided meta data (token symbol, decimals, max supply).
+///
+/// The basic faucet interface exposes two procedures:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// `distribute` requires authentication. The authentication procedure is defined by the specified
+/// authentication scheme.
+pub fn create_basic_faucet(
```

Comment:
I would add to the comment that authentication info is stored in slot 0 and metadata is stored in slot 1.

Reply:
done. Quick question I need to reverse the array when injecting it, right? 

So we store the faucet metadata in account storage at position 1 as `[token_symbol, decimals, 0, max_supply]`. The contract assumes that it gets max_supply by `mem_loadw drop drop drop`. 

So I create the meta_data in reversed order as. Then, when I load the `Word` the stack looks like `[token_symbol, decimals, 0, max_supply]`

```
let meta_data =
        [max_supply, ZERO, Felt::from(decimals), encode_symbol_to_felt(&symbol).unwrap()];
```

Reply:
If the data is stored as `[a0, a1, a2, a3]`, when we load it onto the stack using `mem_loadw`, it will appear on the stack in reverse order - i.e., `[a3, a2, a1, a0, ...]`.

So, I think `[token_symbol, decimals, 0, max_supply]` is correct - but once we load it onto the stack `max_supply` will be at the top of the stack. Should we update the MASM code to account for this?

Reply:
will do